wzdx/experimental_combination/attenuator.py

```diff
@@ -0,0 +1,161 @@
+                wzdx, coordinates[-1][1], coordinates[-1][0], 'route_details_end')
+
+            if not wzdx.get('route_details_end'):
+                logging.warn(
+                    f"Unable to retrieve start point route details for event {wzdx['features'][0].get('id')}")
+                continue
+            if (wzdx['route_details_start']['Route'] != wzdx['route_details_end']['Route']):
+                continue
+
+            for geotab in matching_geotab_routes:
+                logging.debug("Mile markers. geotab: {}, wzdx start: {}, wzdx_end: {}".format(
+                    geotab['route_details']['Measure'], wzdx['route_details_start']['Measure'], wzdx['route_details_end']['Measure']))
+                if wzdx['route_details_start']['Measure'] >= geotab['route_details']['Measure'] and wzdx['route_details_end']['Measure'] <= geotab['route_details']['Measure']:
+                    matching_routes.append((geotab, wzdx))
+                    return matching_routes
+                elif wzdx['route_details_start']['Measure'] <= geotab['route_details']['Measure'] and wzdx['route_details_end']['Measure'] >= geotab['route_details']['Measure']:
+                    matching_routes.append((geotab, wzdx))
+                    return matching_routes
+
+    return matching_routes
+
+
+def add_route(obj, lat, lng, name='route_details'):
+    route_details = cdot_geospatial_api.get_route_and_measure((lat, lng))
+    obj[name] = route_details
+    return obj
+
+
+def combine_geotab_with_wzdx(geotab_avl, wzdx_wzdx):
+    wzdx_wzdx_feature = wzdx_wzdx['features'][0]
+    speed = geotab_avl['avl_location']['position']['speed']
+    bearing = geotab_avl['avl_location']['position']['bearing']
+    route_details = geotab_avl['route_details']
+    distance_ahead = get_distance_ahead_miles(
+        speed, ATTENUATOR_TIME_AHEAD_SECONDS)
+    combined_event = combine_with_wzdx(
+        wzdx_wzdx_feature,
+        route_details,
+        distance_ahead,
+        bearing,
+        wzdx_wzdx['route_details_start']['Measure'],
+        wzdx_wzdx['route_details_end']['Measure'])
+
+    for i in ['route_details', 'route_details_start', 'route_details_end']:
+        if i in combined_event:
+            del combined_event[i]
+    return combined_event
+
+
+def combine_with_wzdx(wzdx_wzdx_feature, route_details, distance_ahead, bearing, event_start_marker, event_end_marker):
+    mmin = min(event_start_marker, event_end_marker)
+    mmax = max(event_start_marker, event_end_marker)
+    geometry, startMarker, endMarker = get_geometry_for_distance_ahead(
+        distance_ahead, route_details, bearing, mmin, mmax)
+    wzdx_wzdx_feature['properties']['beginning_milepost'] = startMarker
+    wzdx_wzdx_feature['properties']['ending_milepost'] = endMarker
+    wzdx_wzdx_feature['geometry']['coordinates'] = geometry
+
+    return wzdx_wzdx_feature
+
+
+def get_geometry_for_distance_ahead(distance_ahead, route_details, bearing, mmin, mmax):
```

Comment:
Looks like we've just mocked this one but not tested outright.

Reply:
Added a unit test